cogs/tas2gif.py

```diff
@@ -0,0 +1,131 @@
+from discord.ext import commands
+from discord.ext import tasks
+import discord
+import requests
+from config import celia_home
+import asyncio, aiohttp
+import os
+import subprocess, shutil
+
+gif_processing = False
+
+class tas2gif(commands.Cog):
+
+    def __init__(self, bot):
+        self.bot = bot
+
+    @commands.command()
+    async def tas2gif(self, ctx, *args):
+
+        global gif_processing
+        if (gif_processing): 
+            await ctx.send('A GIF is currently processing, try again later.')
+            return 
+        gif_processing = True
+
+        if len(args) not in (2, 3):
+            await tas2gif_error(f'Usage: !tas2gif <cartname> <level_index> [url]', ctx)
+            return
+
+        cartname = args[0]
+        level = args[1]
+
+        # Download file
+        attachment_url =  args[2] if len(args) > 2  else ctx.message.attachments[0].url
```

Comment:
Not having an attachment or a url causes the command to fail silently (and worse, gif_processing stays True forever)